contrib/test.sh

```diff
@@ -18,4 +19,22 @@ if [ "$DO_WASM" = true ]; then
     exit 0
 fi
 
-$REPO_DIR/contrib/_test.sh
+for dep in $DEPS
+do
+    cp "Cargo-$dep.lock" Cargo.lock
+    $REPO_DIR/contrib/_test.sh
+
+    if [ "$dep" = recent ];
+    then
+        # We always test committed dependencies but we want to warn if they could've been updated
+        cargo update
+        if diff Cargo-recent.lock Cargo.lock;
+        then
+            echo "Dependencies are up to date"
+        else
+            echo "::warning file=Cargo-recent.lock::Dependencies could be updated"
```

Comment:
In f4078f88:

This warning is harmless enough, but "`cargo update` yields later dependencies" is neither necessary nor sufficient for us to want to update Cargo-recent.lock. I expect this warning to literally always trigger because it finds later dependencies that are incompatible with our MSRV.

Reply:
It only runs for "recent" but I agree, and thought at the time that we added it to `rust-bitcoin` that it was not that useful. Didn't seem to hurt though, and since its informational I acked it. I'm happy to remove it everywhere but I'd prefer to wait till @Kixunil is back because he added it.

Reply:
You have to open the file in GH to see the warning so it at least won't be intrusive.